src/main/java/org/example/repository/ShoppingCartRepository.java

```diff
@@ -0,0 +1,13 @@
+package org.example.repository;
+
+import org.example.model.ShoppingCart;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import java.util.Optional;
+
+@Repository
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+    @EntityGraph(attributePaths = {"cartItems", "cartItems.book", "user"})
```

Comment:
```suggestion
    @EntityGraph(attributePaths = {"cartItems.book", "user"})
```
this should be enough